_posts/2025-03-26-phased-strimzi-upgrade-example.md

```diff
@@ -0,0 +1,702 @@
+
+```mermaid
+flowchart TD
+	S_043(Strimzi 0.43) --> K_A(Kafka A)
+  S_043 --> K_B(Kafka B)
+  S_043 --> K_C(Kafka C)
+	S_044(Strimzi 0.44)
+```
+
+We now want to upgrade to a newer Strimzi version. 
+The Kafka C cluster is our most risk tolerant cluster, it hosts topics used by low tier data pipelines which can tolerate extended downtime, so it will go first to ensure there are no issues.
+We first check that the Kafka C cluster is error free and in the `Ready` state and then apply the Strimzi custom resource selector label for the 0.44.0 Strimzi operator:
+
+```shell
+kubectl -n kafka label --overwrite kafka kafka-c strimzi-resource-selector=strimzi-0-44-0
+```
+
+The 0.44.0 Operator instance will see the new cluster and update the image reference.
+It will then roll the cluster, one broker at a time, on to the new image.
+
+```mermaid
+flowchart TD
+  S_043(Strimzi 0.43) --> K_A(Kafka A)
+  S_043 --> K_B(Kafka B)
+	S_044(Strimzi 0.44) --> K_C(Kafka C)
+```
+
+After a burn-in period, we are happy with the stability of the cluster and no client teams are reporting issues. 
+So we begin to move the rest of the fleet.
+
+```mermaid
+flowchart TD
+	S_043(Strimzi 0.43)
+  S_044(Strimzi 0.44) --> K_A(Kafka A)
+  S_044 --> K_B(Kafka B)
+	S_044 --> K_C(Kafka C)
+	style K_A stroke:#f66,stroke-width:2px,stroke-dasharray: 5 5
+```
+
+All is well until we move our highest tier cluster, A. 
+The move to Strimzi 0.44 revealed a configuration issue.
+We think we know what the issue is and how to fix it, but as this is a critical cluster we need to roll back and make sure all the dependant services are running correctly. 
+In this setup the roll back is a simple label change:
+
+```shell
+kubectl -n kafka label --overwrite kafka kafka-a strimzi-resource-selector=strimzi-0-43-0
+```
+
+```mermaid
+flowchart TD
+  S_043(Strimzi 0.43) --> K_A(Kafka A)
+  S_044(Strimzi 0.44) --> K_B(Kafka B)
+	S_044 --> K_C(Kafka C)
+```
+
+We fix the configuration issue and move Kafka cluster A over to the 0.44 operator.
+
+```mermaid
+flowchart TD
+  S_043(Strimzi 0.43)
+	style S_043 stroke:#f66,stroke-width:2px
+  S_044(Strimzi 0.44) --> K_A(Kafka A)
+  S_044 --> K_B(Kafka B)
+	S_044 --> K_C(Kafka C)
+```
+
+At this point the Strimzi 0.43 operator is not managing any Kafka clusters and all the clusters being managed by the 0.44 operator have been stable for a sufficient period that we are confident we don't need to roll back. 
+We can now remove the Strimzi 0.43 operator deployment and its associated RBAC resources.
+At this point we can add a new version to the Operator Set, adding the 0.45 installation files and upgrading the CRDs to those of the 0.45 version.
+
+```mermaid
+flowchart TD
+  S_044(Strimzi 0.44) --> K_A(Kafka A)
+  S_044 --> K_B(Kafka B)
+	S_044 --> K_C(Kafka C)
+  S_045(Strimzi 0.45)
+	style S_045 stroke:#008000,stroke-width:2px
+```
+
```

Comment:
Maybe you can here at the end mention that the same process can be used for Connect, MM2, and Bridge?